.github/workflows/pipeline.yml

```
@@ -0,0 +1,169 @@
+          cd artifacts
+          tar -xzf backend-deployment.tar.gz
+          
+          # Upload to S3 bucket
+          aws s3 sync . s3://${{ secrets.S3_BUCKET_NAME }}/backend/ \
+            --delete \
+            --exclude "*.tar.gz" \
+            --cache-control "max-age=31536000,public" \
+            --metadata-directive REPLACE
+
+      - name: Invalidate CloudFront cache
+        if: ${{ secrets.CLOUDFRONT_DISTRIBUTION_ID }}
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)